packages/slate-editor/src/modules/editor-v4-floating-add-menu/lib/shouldShowMenuButton.ts

```diff
@@ -17,6 +17,9 @@ export function shouldShowMenuButton(editor: Editor): boolean {
     }
 
     const text = Node.string(currentNode);
+    const hasPlaceholder = currentNode.children.some(
```

Comment:
I suggest generalizing the check and instead of looking for placeholder nodes, consider a paragraph as non-empty if there are any void children (like placeholder).

From practical point of view, the original code can be modified to:

```diff
- if (!isParagraphNode(currentNode)) {
+ if (!isParagraphNode(currentNode) || hasVoidElements(editor, currentNode)) {
    return false;
  }
```

`hasVoidElements()` is already implemented in *isEmptyParagraphElement.ts* (so you'll need to extract it to a separate exported function in the `commands` folder).

Reply:
Hmm. I don't see anything like that: 
![Screenshot 2022-04-05 at 15 34 06](https://user-images.githubusercontent.com/41072665/161765991-4f9c7ec8-883f-4120-b2e6-d68b0f24ea29.png)

Reply:
https://github.com/prezly/slate/blob/master/packages/slate-commons/src/commands/isEmptyParagraphElement.ts

Reply:
done